### PR TITLE
feat: add get next active slot api endpoint

### PR DESCRIPTION
**Pull request by vladimir-trifonov**
feat: add get next active slot api endpoint

### CODE REVIEW

File: crates/api/src/gossiper/grpc_gossiper.rs [hunk 1]
```diff
@@ -164,7 +164,7 @@ impl GrpcGossiperClientManager {
     pub async fn start_server(&self, builder_api_sender: Sender<GossipedMessage>, proposer_api_sender: Sender<GossipedMessage>) {
         let service = GrpcGossiperService { builder_api_sender, proposer_api_sender };
 
-        let addr = "0.0.0.0:50051".parse().unwrap();
+        let addr = "0.0.0.0:50055".parse().unwrap();
```
Comment (failfmi) [inline crates/api/src/gossiper/grpc_gossiper.rs]: is this change needed?

File: crates/api/src/proposer/api.rs [hunk 2]
```diff
@@ -785,6 +785,46 @@ where
         Ok(get_payload_response)
     }
 
+    pub async fn get_next_active_slot(
```
Comment (failfmi) [inline crates/api/src/proposer/api.rs:788]: let's discuss this

File: crates/api/src/gossiper/grpc_gossiper.rs (hunk 1 above)
Comment (vladimir-trifonov) [reply crates/api/src/gossiper/grpc_gossiper.rs]: currently for our setup, yes, it says the port is used otherwise, not sure what is running on it

Comment (vladimir-trifonov) [reply crates/api/src/gossiper/grpc_gossiper.rs]: nvm, it works without it

File: crates/api/src/proposer/api.rs (hunk 2 above)
Comment (vladimir-trifonov) [reply crates/api/src/proposer/api.rs:788]: this remains, right? since we are taking the slot to validate it in the commit-boost?

Comment (failfmi) [reply crates/api/src/proposer/api.rs:788]: I thought about adding the slot to the `/conditions` endpoint and excluding this call.

**Review by failfmi — APPROVED** [review]
